corehq/apps/userreports/reports/view.py

```diff
@@ -124,8 +125,12 @@ def domain(self):
     @use_nvd3
     @conditionally_location_safe(has_location_filter)
     def dispatch(self, request, *args, **kwargs):
-        original = super(ConfigurableReport, self).dispatch(request, *args, **kwargs)
-        return original
+        if self.spec.report_meta.created_by_builder and has_report_builder_access(request):
+            original = super(ConfigurableReport, self).dispatch(request, *args, **kwargs)
+            return original
+        else:
+            from corehq.apps.userreports.views import paywall_home
+            return HttpResponseRedirect(paywall_home(self.domain))
```

Comment:
I think this is causing non-report builder UCRs to all be redirected to the report builder paywall. This view is used for both hand built UCRs and report builder UCRs, so the hand built ones fail `self.spec.report_meta.created_by_builder`, which result in the user getting sent to the paywall.
@kaapstorm I believe this is the issue that you are trying to solve in https://github.com/dimagi/commcare-hq/pull/14678